java/source/io.rst

```diff
@@ -443,3 +443,109 @@ Reading Parquet File
+        countries.set(5, "Malta".getBytes(StandardCharsets.UTF_8));
+        countries.set(6, "Tailandia".getBytes(StandardCharsets.UTF_8));
+        countries.set(7, "Uganda".getBytes(StandardCharsets.UTF_8));
+        countries.set(8, "Yemen".getBytes(StandardCharsets.UTF_8));
+        countries.set(9, "Zambia".getBytes(StandardCharsets.UTF_8));
+        countries.setValueCount(10);
+
+        Dictionary countriesDictionary = new Dictionary(countries,
+                new DictionaryEncoding(/*id=*/123L, /*ordered=*/false, /*indexType=*/new ArrowType.Int(8, true)));
+        System.out.println("Dictionary: " + countriesDictionary);
+
+        appUserCountriesUnencoded.allocateNew(5);
+        appUserCountriesUnencoded.set(0, "Andorra".getBytes(StandardCharsets.UTF_8));
+        appUserCountriesUnencoded.set(1, "Guinea".getBytes(StandardCharsets.UTF_8));
+        appUserCountriesUnencoded.set(2, "Islandia".getBytes(StandardCharsets.UTF_8));
+        appUserCountriesUnencoded.set(3, "Malta".getBytes(StandardCharsets.UTF_8));
+        appUserCountriesUnencoded.set(4, "Uganda".getBytes(StandardCharsets.UTF_8));
+        appUserCountriesUnencoded.setValueCount(5);
+        System.out.println("Unencoded data: " + appUserCountriesUnencoded);
+
+        File file = new File("random_access_file_with_dictionary.arrow");
+        DictionaryProvider.MapDictionaryProvider provider = new DictionaryProvider.MapDictionaryProvider();
+        provider.put(countriesDictionary);
+        try (FieldVector appUseCountryDictionaryEncoded = (FieldVector) DictionaryEncoder
+                .encode(appUserCountriesUnencoded, countriesDictionary);
+             VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.of(appUseCountryDictionaryEncoded);
+             FileOutputStream fileOutputStream = new FileOutputStream(file);
+             ArrowFileWriter writer = new ArrowFileWriter(vectorSchemaRoot, provider, fileOutputStream.getChannel())
+        ) {
+            System.out.println("Dictionary-encoded data: " +appUseCountryDictionaryEncoded);
+            writer.start();
+            writer.writeBatch();
+            writer.end();
+            System.out.println("Record batches written: " + writer.getRecordBlocks().size() + ". Number of rows written: " + vectorSchemaRoot.getRowCount());
+            try(
+                    BufferAllocator rootAllocator = new RootAllocator();
+                    FileInputStream fileInputStream = new FileInputStream(file);
+                    ArrowFileReader reader = new ArrowFileReader(fileInputStream.getChannel(), rootAllocator)
+            ){
+                for (ArrowBlock arrowBlock : reader.getRecordBlocks()) {
+                    reader.loadRecordBatch(arrowBlock);
+                    FieldVector appUseCountryDictionaryEncodedRead = reader.getVectorSchemaRoot().getVector("app-use-country-dict");
+                    Dictionary appUseCountryDictionaryRead = reader.getDictionaryVectors().get(123L);
```

Comment:
Hmm, isn't there some way to associate the column with the dictionary ID so that you don't have to hardcode this? Though, I don't see it from a quick glance through some tests/docs so maybe not in Java.

Reply:
No, that's not what I mean; in C++ the dictionary IDs are associated with the columns somehow so that you don't have to manually hardcode the dictionary IDs

Reply:
The dictionary ID should be somehow settable on the Field and then we don't have to hard code this here: https://github.com/apache/arrow/blob/7adda73dc4ef6f28f8e5701eb4be1dc9526c0e1b/format/Schema.fbs#L473-L474

Reply:
Yeah, you can attach a dictionary ID in FieldType: https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/types/pojo/FieldType.html

Reply:
Changed